code-build.js

```diff
@@ -50,7 +57,33 @@ async function waitForBuildEndTime(sdk, { id, logs }, nextToken) {
       cloudWatchLogs
         .getLogEvents({ logGroupName, logStreamName, startFromHead, nextToken })
         .promise()
-  ]);
+  ]).catch(err => {
+    errObject = err;
+
```

Comment:
```suggestion
/* Returning [] here so that the assignment above
 * does not throw `TypeError: undefined is not iterable`.
 * The error is handled below,
 * since it might be a rate limit.
 */
```